pipelined/bevy_render2/src/render_resource/shader.rs

```diff
@@ -268,10 +268,10 @@ impl ShaderProcessor {
         for line in shader.split('\n') {
             if let Some(cap) = self.ifdef_regex.captures(line) {
                 let def = cap.get(1).unwrap();
-                scopes.push(shader_defs.contains(def.as_str()));
+                scopes.push(*scopes.last().unwrap() && shader_defs.contains(def.as_str()));
```

Comment:
Can you add another unit test to the bottom that covers nesting? This is the kind of feature that should have tests backing it up.

Reply:
Done, for the 4 cases.